server/sonar-db-dao/src/main/java/org/sonar/db/qualitygate/QualityGateDao.java

```diff
@@ -59,6 +61,12 @@ public void ensureOneBuiltInQualityGate(DbSession dbSession, String builtInName)
     mapper(dbSession).ensureOneBuiltInQualityGate(builtInName);
   }
 
+  public QualityGateDto selectBuiltIn(DbSession dbSession) {
+    QualityGateDto qualityGateDto = mapper(dbSession).selectBuiltIn();
+    checkState(qualityGateDto != null, "Builtin quality gate is missing.");
```

Comment:
AFAIK, such check should not be in DAO (it probably should be in QualityGateFinder).